packages/beacon-node/src/chain/archiver/index.ts

```diff
@@ -143,4 +166,165 @@ export class Archiver {
+      .map((indexString) => Number(indexString));
+    const finalizedAttachedValidatorsCount = attachedProposers.length;
+
+    // Calculate stats for attached validators, based on states in checkpointState cache
+    let finalizedFoundCheckpointsInStateCache = 0;
+
+    let expectedAttachedValidatorsProposalsCount = 0;
+    let finalizedAttachedValidatorsProposalsCount = 0;
+    let finalizedAttachedValidatorsOrphanCount = 0;
+    let finalizedAttachedValidatorsMissedCount = 0;
+
+    for (const checkpointHex of finalizedProposersCheckpoints) {
+      const checkpointState = checkpointStateCache.get(checkpointHex);
+
+      // Generate stats for attached validators if we have state info
+      if (checkpointState !== null) {
+        finalizedFoundCheckpointsInStateCache++;
+
+        const epochProposers = checkpointState.epochCtx.proposers;
+        const startSlot = checkpointState.epochCtx.epoch * SLOTS_PER_EPOCH;
+
+        for (let index = 0; index < epochProposers.length; index++) {
+          const slot = startSlot + index;
+
+          // Let skip processing the slots which are out of range
+          // Range to consider is:
+          //   lastFinalized.epoch * SLOTS_PER_EPOCH + 1, .... finalized.epoch * SLOTS_PER_EPOCH
+          if (slot <= lastFinalized.epoch * SLOTS_PER_EPOCH || slot > finalized.epoch * SLOTS_PER_EPOCH) {
+            continue;
+          }
+
+          const proposer = epochProposers[index];
+
+          // If this proposer was attached to this BN for this epoch
+          if (attachedProposers.includes(proposer)) {
+            expectedAttachedValidatorsProposalsCount++;
+
+            // Get what validators made canonical/non canonical proposals for this slot
+            const {canonicalVals, nonCanonicalVals} = slotProposers.get(slot) ?? {
+              canonicalVals: [],
+              nonCanonicalVals: [],
+            };
+            let wasFinalized = false;
+
+            if (canonicalVals.includes(proposer)) {
+              finalizedAttachedValidatorsProposalsCount++;
+              wasFinalized = true;
+            }
+            const attachedProposerNonCanSlotProposals = nonCanonicalVals.filter((nonCanVal) => nonCanVal === proposer);
+            finalizedAttachedValidatorsOrphanCount += attachedProposerNonCanSlotProposals.length;
+
+            // Check is this slot proposal was missed by this attached validator
+            if (!wasFinalized && attachedProposerNonCanSlotProposals.length === 0) {
+              finalizedAttachedValidatorsMissedCount++;
+            }
+          }
+        }
+      }
+    }
+
+    const attachedValidators: ProposalStats = {
+      expected: expectedAttachedValidatorsProposalsCount,
+      finalized: finalizedAttachedValidatorsProposalsCount,
+      orphaned: finalizedAttachedValidatorsOrphanCount,
+      missed: finalizedAttachedValidatorsMissedCount,
+    };
+
+    this.logger.info("All validators finalized stats", {
```

Comment:
it this something we always want to log to info?

Reply:
finalized stats people should find useful in normal usage